src/functional/program_processors/codegens/gtfn/gtfn_ir.py

```diff
@@ -53,6 +53,11 @@ class TernaryExpr(Expr):
     false_expr: Expr
 
 
+class CastExpr(Expr):
```

Comment:
This is a valid way to the express a `cast` in an IR (@petiaccja actually asked me a while ago why we don't express more information in the types of the IR nodes the same way you did here). However it is not in line with the rest of the IR and for consistency I suggest the `cast` to be represented as a regular `FunCall` node to the `builtin` function `cast`.

Reply:
I am asking this for personal understanding, but I don't think that we should do:

```FunCall = as_fmt("static_cast<{new_dtype}>({obj_expr})")```

Perhaps it is not even necessary under these circumstances (?)

Reply:
I think the central problem is that `FunCall` has a `list[Expr]` as arguments, so you cannot pass a dtype there.

The options I see:
1. `FunCall(fun=SymRef(id="cast"), args=[TypeLiteral(dtype=float), <value>])`
2. `CastExpr(type=float, value=<value>)`
3. `FunCall(fun=SymRef(id="cast"), args=[SymRef(id="float"), <value>])`
4. `FunCall(fun=SymRef(id="float"), args=[<value>])`

For number 3) and 4), you have to add supported types as builtin symbols. IMHO we can drop 3), it's just a more complicated version of 4), and also makes the type of the new builtins questionable, because with 3) they are clearly functions.

If you want to represent types in the IR by other than strings, I believe you have to add a new node. It could be either a `TypeLiteral` or a `CastExpr`. If you use a `cast` builtin function, then there is one more special casing for the `FunCall` node and there is one more constraint (1st argument must be `TypeLiteral`) that is not verified by static analysis. `CastExpr` doesn't add new special cases or constraints. On the other hand, using a `cast` builtin is more in line with the rest of the IR.

Just my 2 cents.